### ISSUE TITLE
Unifi UXG Pro (UniFi Gateway Pro) Cannot Start / Error

### ISSUE THREAD

**Issue by zanelucas** [Control-D-Inc/ctrld#71]
**On the UniFi Gateway Pro I am running the following command to install which is successful :**

# sh -c 'sh -c "$(curl -sL https://api.controld.com/dl)"'

---------------------
|    System Info    |
---------------------
OS Type      : linux
OS Vendor    : ubios
OS Version   : 3.1.8.12135
Router Model : UniFi NeXt-Gen Gateway PRO
Arch         : aarch64
CPU          : Cortex-A57
Free RAM     : 323 MB / 1997 MB
---------------------
|  Install Details  |
---------------------
Binary URL   : https://assets.controld.com/ctrld/linux/arm64/ctrld
Install Path : /data/controld
---------------------
Install binary and run it? (y/n): y
 - Starting download
 - Making binary executable
 - Launching /data/controld/ctrld
---------------------

**However, viewing the config file located in /etc/controld/ctrld.toml the default port seems to still be setup (not 5354 as per the discussions) :** 

[listener]
  [listener.0]
    ip = '127.0.0.1'
    port = 53

**After changing the port to 5354 and attempting to start the service I am seeing the following error :** 

failed to get default route interface error="defaultRouteFromNetlink: Dial: setsockopt: protocol not available

**Are there any ideas on what can be tried next, or is this router type not being setup correctly?**

**Comment by yegors** [comment]
Are you using it with a ctrld upstream? If so, you should run the full installer command from the web panel (after creating a Ubiqiti Device), which includes your resolver ID in the command flags. This will start the service automatically and template the correct things into the config. 

If you're editing the config manually, you're responsible for resolving listener conflicts.

**Comment by zanelucas** [comment]
At this point I am only using the default configuration (which uses ctrld upstreams by default). I was following as per the details : 

Supported Routers

You can run ctrld on any supported router, which will function similarly to the Service Mode mentioned above. The list of supported routers and firmware includes:

......

ctrld will attempt to interface with dnsmasq whenever possible and set itself as the upstream, while running on port 5354. On FreeBSD based OSes, ctrld will terminate dnsmasq and unbound in order to be able to listen on port 53 directly.


**It was not clear that in order for this to actually work by default that the next section in the docs "Control D Auto Configuration" needed to also be done. I assumed that when attempting to start the service that it would create a basic configuration that would work, not a broken one (like when running on a machine) - that's not the case now that I have read your comment.**

**Comment by yegors** [comment]
I do agree the default generated config should work "out of the box".  We will make it so in the next update.

That being said, editing the default config to use port 5354 (or any other non-conflicting port) will make it work on a **default USG install with no conflicting software running**. 

```
root@ubnt:~# host dolohen.com
dolohen.com has address 139.45.197.236
root@ubnt:~# cat /etc/controld/ctrld.toml 
[listener]
  [listener.0]
    ip = '127.0.0.1'
    port = 12345

[network]
  [network.0]
    name = 'Network 0'
    cidrs = ['0.0.0.0/0']

[upstream]
  [upstream.0]
    name = 'Control D - Anti-Malware'
    type = 'doh'
    endpoint = 'https://freedns.controld.com/p1'
    bootstrap_ip = '76.76.2.11'
    timeout = 5000

  [upstream.1]
    name = 'Control D - No Ads'
    type = 'doq'
    endpoint = 'p2.freedns.controld.com'
    bootstrap_ip = '76.76.2.11'
    timeout = 3000
root@ubnt:~# ctrld start -vv
Aug 16 16:41:14.000 INF loading config file from: /etc/controld/ctrld.toml
Aug 16 16:41:14.000 DBG cleaning up router before installing
Aug 16 16:41:14.000 NTC Starting service
Aug 16 16:41:14.000 DBG waiting for ctrld listener to be ready
Aug 16 16:41:22.000 DBG ctrld listener is ready
Aug 16 16:41:22.000 DBG performing self-check
Aug 16 16:41:22.000 DBG self-check against "verify.controld.com" succeeded
Aug 16 16:41:22.000 NTC Service started
Aug 16 16:41:22.000 DBG NetworkManager is not available
Aug 16 16:41:22.000 DBG setting DNS for interface iface=eth0
Aug 16 16:41:22.000 DBG dns: [rc=unknown ret=direct]
Aug 16 16:41:22.000 DBG dns: using "direct" mode
Aug 16 16:41:23.000 DBG setting DNS successfully iface=eth0
root@ubnt:~# host dolohen.com
dolohen.com has address 0.0.0.0
dolohen.com has IPv6 address ::
root@ubnt:~# netstat -tupln
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      2900/lighttpd   
tcp        0      0 0.0.0.0:39443           0.0.0.0:*               LISTEN      2931/redirector 
tcp        0      0 0.0.0.0:53              0.0.0.0:*               LISTEN      30649/dnsmasq   
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      3083/sshd       
tcp        0      0 127.0.0.1:12345         0.0.0.0:*               LISTEN      29821/ctrld     
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      2900/lighttpd   
tcp        0      0 127.0.0.101:55523       0.0.0.0:*               LISTEN      2896/telnetd    
tcp        0      0 0.0.0.0:39080           0.0.0.0:*               LISTEN      2931/redirector 
tcp6       0      0 :::80                   :::*                    LISTEN      2900/lighttpd   
tcp6       0      0 :::53                   :::*                    LISTEN      30649/dnsmasq   
tcp6       0      0 :::22                   :::*                    LISTEN      3083/sshd       
tcp6       0      0 :::443                  :::*                    LISTEN      2900/lighttpd   
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           29821/ctrld     
udp        0      0 0.0.0.0:5353            0.0.0.0:*                           29821/ctrld     
udp        0      0 0.0.0.0:41489           0.0.0.0:*                           737/rsyslogd    
udp        0      0 0.0.0.0:10001           0.0.0.0:*                           2841/mcad       
udp        0      0 0.0.0.0:1812            0.0.0.0:*                           715/freeradius  
udp        0      0 0.0.0.0:1813            0.0.0.0:*                           715/freeradius  
udp        0      0 0.0.0.0:1814            0.0.0.0:*                           715/freeradius  
udp        0      0 0.0.0.0:53              0.0.0.0:*                           30649/dnsmasq   
udp        0      0 127.0.0.1:12345         0.0.0.0:*                           29821/ctrld     
udp        0      0 0.0.0.0:67              0.0.0.0:*                           2613/dhcpd3     
udp        0      0 0.0.0.0:68              0.0.0.0:*                           2960/dhclient   
udp        0      0 0.0.0.0:39766           0.0.0.0:*                           2841/mcad       
udp        0      0 192.168.1.1:123         0.0.0.0:*                           2537/ntpd       
udp        0      0 10.0.21.88:123          0.0.0.0:*                           2537/ntpd       
udp        0      0 127.0.0.1:123           0.0.0.0:*                           2537/ntpd       
udp        0      0 0.0.0.0:123             0.0.0.0:*                           2537/ntpd       
udp        0      0 0.0.0.0:44451           0.0.0.0:*                           715/freeradius  
udp        0      0 127.0.0.1:18120         0.0.0.0:*                           715/freeradius  
udp6       0      0 :::53                   :::*                                30649/dnsmasq   
udp6       0      0 fe80::72a7:41ff:fef:123 :::*                                2537/ntpd       
udp6       0      0 ::1:123                 :::*                                2537/ntpd       
udp6       0      0 fe80::72a7:41ff:fef:123 :::*                                2537/ntpd       
udp6       0      0 :::123                  :::*                                2537/ntpd       
root@ubnt:~# 
```

**Comment by yegors** [comment]
Hi,

Please try the new dev build, it should fix the issue. Default rendered config will now work as is. Use the installer below to download the dev build. 

```
sh -c 'sh -c "$(curl -sSL https://api.controld.dev/dl)"'
```

**Comment by cuonglm** [comment]
This should be fixed in v1.3.1 release.
